specs/execution-api.md

```diff
@@ -0,0 +1,74 @@
+# Execution API Specification
+
+## Overview
+
+The Execution API is the interface `Conductor` uses to drive deterministic execution of sequenced blocks in dependent chains. Inspired by other APIs, such as the Engine API and ABCI Consensus API, it is a chain agnostic mechanism intended to be very simple to implement. It is a gRPC API which any state machine can implement, and use conductor with to drive their execution and integrate with the Astria Sequencer.
+
+> Note: this documentation is for the v1alpha2 API which is currently being implemented, and some of the documentation is on how it should be implemented not how it is.
+
+## Basic Design Principles
+
+The Execution API is a resource based API with two resources: `Block` and `CommitmentState`. The API is designed to follow basic principles outlined by aip.dev as best practices for resource based APIs. gRPC has been chosen for the API due to the wide availability of language implementations which make it easy to generate client libraries, as well as stubs for server implementations.  
+
+## Conductor Usage
+
+### Startup
+
+Upon startup, conductor needs to know the state of commitments in the state machine and calls `GetCommitmentState`. If started on a fresh rollup these will all be the same block. If running against a state machine with previous block data, Conductor must also track the block hash of any blocks between commitments, it will call `BatchGetBlock` to get block information between commitments.
+
+### Execution & Commitments
+
+From the perspective of the sequencer, `HEAD` commitments are made everytime the sequencer creates a new block at height N. The transactions on this block may change up until sequencer block at height N+1 has been created, at this point the block at height N has a `SOFT` commitment. A soft commitment means that sequencer consensus has full agreement on the block. Once a block has been seen on DA it has a `FIRM` commitment, it has been made widely available across the DA network.
+
+When configuring conductor, you can configure the time at which blocks are executed in your rollup using the `execution_commitment_level` in the config file. If this is configured to a higher level of commitment, no action will be taken upon receiving lower commitments. 
```

Comment:
`CreateBlockRequest` doesn't specify the commitment level so I assume will always set the latest block as `HEAD`. so if `execution_commitment_level = soft` for example, then the conductor will send a `CreateBlockRequest` on receiving a soft commit and then immediately set the just executed block to `SOFT`?

Reply:
I will clarify in the `CreateBlock` section but as seen in the sequence diagram, CreateBlock creates a block but it does not mark it as the head, this is done in a follow up request. This is a departure from our previous API. This is done, in part to enable this different execution (it also follows a pattern similar to Engine API)